### PR TITLE
New IngestDecode hidden stage (breaking change)

### PR DESCRIPTION
**Pull request by jotak**
A new IngestDecode stage is added, that produces GenericMap output.
This stage is almost transparent to the users, as they'll continue to use
"ingest" like a usual ingester, but without a following "decode" stage. Impacted ingesters are
Collector and GRPC (which always decodes protobuf)

### CODE REVIEW

**Review by mariomac — COMMENTED** [review]
good improvement. Thanks!

**Review by mariomac — APPROVED** [review]

File: pkg/pipeline/ingest_decode/ingest_collector.go [hunk 1]
```diff
@@ -109,7 +108,7 @@ func (w *TransportWrapper) Send(_, data []byte) error {
 }
 
 // Ingest ingests entries from a network collector using goflow2 library (https://github.com/netsampler/goflow2)
-func (ingestC *ingestCollector) Ingest(out chan<- []interface{}) {
+func (ingestC *ingestCollector) Ingest(out chan<- []config.GenericMap) {
 	ctx := context.Background()
```
Comment (KalmanMeth) [inline pkg/pipeline/ingest_decode/ingest_collector.go:112]: Should we change the name of the interface function (Ingest) so we don't confuse it with an Ingestor? Or do we now dispose with the distinction between stage names, and any first stage is called Ingest?

File: pkg/pipeline/ingest_decode/ingest_decode.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ *
+ */
+
+package ingest_decode
+
+import "github.com/netobserv/flowlogs-pipeline/pkg/config"
+
+type IngesterDecoder interface {
+	Ingest(out chan<- []config.GenericMap)
```
Comment (KalmanMeth) [inline pkg/pipeline/ingest_decode/ingest_decode.go:23]: Perhaps IngestDecode().

**Review by KalmanMeth — APPROVED** [review]

File: pkg/pipeline/ingest_decode/ingest_decode.go (hunk 2 above)
Comment (jotak) [reply pkg/pipeline/ingest_decode/ingest_decode.go]: done

File: pkg/pipeline/ingest_decode/ingest_collector.go (hunk 1 above)
Comment (jotak) [reply pkg/pipeline/ingest_decode/ingest_collector.go]: sure, done

File: pkg/pipeline/ingest_decode/ingest_collector.go [hunk 3]
```diff
@@ -109,7 +108,7 @@ func (w *TransportWrapper) Send(_, data []byte) error {
 }
 
 // Ingest ingests entries from a network collector using goflow2 library (https://github.com/netsampler/goflow2)
```
Comment (eranra) [inline pkg/pipeline/ingest_decode/ingest_collector.go:110]: s/Ingest/IngestDecode

**Comment by jotak** [comment]
closing: superseded by #225

**Pull request closed without merging**